src/linear_algebra/nonsingular_inverse.lean

```diff
@@ -0,0 +1,300 @@
+/-
+  Copyright (c) 2019 Tim Baanen. All rights reserved.
+  Released under Apache 2.0 license as described in the file LICENSE.
+  Author: Tim Baanen.
+
+  Inverses for nonsingular square matrices.
+-/
+import algebra.big_operators
+import data.matrix.basic
+import linear_algebra.determinant
+
+/-!
+# Nonsingular inverses
+
+In this file, we define an inverse for square matrices of invertible
+determinant. For matrices that are not square or not of full rank, there is a
+more general notion of pseudoinverses. Unfortunately, the definition of
+pseudoinverses is typically in terms of inverses of nonsingular matrices, so we
+need to define those first. The file also doesn't define a `has_inv` instance
+for `matrix` so that can be used for the pseudoinverse instead.
+
+The definition of inverse used in this file is the one given by Cramer's rule.
+The vectors returned by Cramer's rule are given by the linear map `cramer`.
+Using Cramer's rule, we can compute for each matrix `A` the matrix `adjugate A`,
+which we then prove behaves like `det A • A⁻¹`. Finally, we show that dividing
+the adjugate by `det A` (if possible), giving a matrix `nonsing_inv A`, will
+result in a multiplicative inverse to `A`.
+
+## References
+
+  * https://en.wikipedia.org/wiki/Cramer's_rule#Finding_inverse_matrix
+
+## Tags
+
+matrix inverse, cramer, cramer's rule, adjugate
+-/
+
+namespace matrix
+universes u v
+variables {n : Type u} [fintype n] [decidable_eq n] {α : Type v}
+open_locale matrix
+open equiv equiv.perm finset
+
+section replace
```

Comment:
Since you are using `function.update`, why not go with `update_column` and so on for coherence?